feat: stacking error list

magnew opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

Closes: #11198 

### Description:

Makes it possible to add errors to a list instead of each in it's own toast. That list is collapsable and the errors can be dismissed individually. 

Note that there is a new method `addToastError` that will add to the list and I kept the old `showToastError` for now. This way we can keep the existing behavior in some places, but use error lists in others. For now I have only updated a few places where multiple errors were most problematic to use the error list, primarily the API error reporting. 

The collapsed errors now look like:
<img width="470" alt="Screenshot 2024-10-30 at 09 58 29" src="https://github.com/user-attachments/assets/b6472a00-fc7e-462a-963a-a81187ecfe15">


And the expanded list looks like:
<img width="466" alt="Screenshot 2024-10-30 at 09 58 44" src="https://github.com/user-attachments/assets/054263b4-ea56-4253-85dd-3019ddc2aa24">

 One consistent way to test this is to add a table calculation with invalid columns or no sql, then do it again. The first toast will be the same as always, the second one will add to the list.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

rephus reviewed (commented):
Nice, this works great, I tested a few errors and it is working as expected. Just a small UI suggestion, could we get rid of the weird scrollbar when there are exactly 3 items ? it seems uneccesary, perhaps by increasing the size of the panel a few pixels. 
[Screencast from 30-10-24 10:58:15.webm](https://github.com/user-attachments/assets/f3bd3d5f-c79e-4d33-a6db-b4bdae192eee)


My suggestion was to reduce the amount of duplicated errors by reusing the `key` in the tooptip, I believe this is what we do in sql runner. But this is a useful approach. 


[Screencast from 30-10-24 11:00:45.webm](https://github.com/user-attachments/assets/f57f1029-a7da-40bf-8137-bfedbb95250e)

magnew commented:
> Nice, this works great, I tested a few errors and it is working as expected. Just a small UI suggestion, could we get rid of the weird scrollbar when there are exactly 3 items ? it seems uneccesary, perhaps by increasing the size of the panel a few pixels.
> 
>  Screencast.from.30-10-24.10.58.15.webm 
> My suggestion was to reduce the amount of duplicated errors by reusing the `key` in the tooptip, I believe this is what we do in sql runner. But this is a useful approach.
> 
>  Screencast.from.30-10-24.11.00.45.webm

Thanks for looking! Yea, let me see if I can make it better in the 3-items case

TuringLovesDeathMetal commented:
niiiiice! Looks great 😍 agreed with Javi - two tiny things: 

1. the scroll when there are three items feels a bit funny

<img width="462" alt="image" src="https://github.com/user-attachments/assets/f61337f1-7782-4522-b305-442e01eee710">

2. when there are two items, I wish it resized to have a fixed margin (instead of having a bunch of empty space)

<img width="477" alt="image" src="https://github.com/user-attachments/assets/0ec31c0c-9548-4161-a6ba-49a5031303db">


If this is going to take more than like an hour to figure out though, then I think we can skip them and open them as a separate issue and go ahead and merge this! ✅

magnew commented:
> niiiiice! Looks great 😍 agreed with Javi - two tiny things:
> 
> 1. the scroll when there are three items feels a bit funny
> 
> <img alt="image" width="462" src="https://private-user-images.githubusercontent.com/31848148/381517779-f61337f1-7782-4522-b305-442e01eee710.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyODY3OTMsIm5iZiI6MTczMDI4NjQ5MywicGF0aCI6Ii8zMTg0ODE0OC8zODE1MTc3NzktZjYxMzM3ZjEtNzc4Mi00NTIyLWIzMDUtNDQyZTAxZWVlNzEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMwVDExMDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjNzZjNTAwYzBiNzYzOTYxZDZkZmVlMzgwYzViMzkxNDM4Y2Q3NDFmY2FkNTkwNzBhNDZlY2EyNzQwYzJkN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bch1Se6Qm4dCoyj3hix17Zug3Kgj3GWQ_UFaxhXvySU">
> 2. when there are two items, I wish it resized to have a fixed margin (instead of having a bunch of empty space)
> 
> <img alt="image" width="477" src="https://private-user-images.githubusercontent.com/31848148/381517633-0ec31c0c-9548-4161-a6ba-49a5031303db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyODY3OTMsIm5iZiI6MTczMDI4NjQ5MywicGF0aCI6Ii8zMTg0ODE0OC8zODE1MTc2MzMtMGVjMzFjMGMtOTU0OC00MTYxLWE2YmEtNDlhNTAzMTMwM2RiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMwVDExMDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYWUxNzQwZGIwMzk2YjQ5MGE3M2Q1MjYxNmU3ZTA3ZmNkZWU4NTMyMDc0Yjc2MjI2ZTM0OTIwNjZiNDA1NGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l6DyPCUBmXm1ZXLiwtUt3rbLapBN5MEtjlfFx4DzYKw">
> If this is going to take more than like an hour to figure out though, then I think we can skip them and open them as a separate issue and go ahead and merge this! ✅

I fixed part of this. Getting the thing to grow vertically, then scroll has been a bit problematic, but I'll spend a few more minutes. I agree it looks dumb.

magnew commented:
> > niiiiice! Looks great 😍 agreed with Javi - two tiny things:
> > 
> > 1. the scroll when there are three items feels a bit funny
> > 
> > <img alt="image" width="462" src="https://private-user-images.githubusercontent.com/31848148/381517779-f61337f1-7782-4522-b305-442e01eee710.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyODY3OTMsIm5iZiI6MTczMDI4NjQ5MywicGF0aCI6Ii8zMTg0ODE0OC8zODE1MTc3NzktZjYxMzM3ZjEtNzc4Mi00NTIyLWIzMDUtNDQyZTAxZWVlNzEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMwVDExMDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjNzZjNTAwYzBiNzYzOTYxZDZkZmVlMzgwYzViMzkxNDM4Y2Q3NDFmY2FkNTkwNzBhNDZlY2EyNzQwYzJkN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bch1Se6Qm4dCoyj3hix17Zug3Kgj3GWQ_UFaxhXvySU">
> > 2. when there are two items, I wish it resized to have a fixed margin (instead of having a bunch of empty space)
> > <img alt="image" width="477" src="https://private-user-images.githubusercontent.com/31848148/381517633-0ec31c0c-9548-4161-a6ba-49a5031303db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyODY3OTMsIm5iZiI6MTczMDI4NjQ5MywicGF0aCI6Ii8zMTg0ODE0OC8zODE1MTc2MzMtMGVjMzFjMGMtOTU0OC00MTYxLWE2YmEtNDlhNTAzMTMwM2RiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMwVDExMDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYWUxNzQwZGIwMzk2YjQ5MGE3M2Q1MjYxNmU3ZTA3ZmNkZWU4NTMyMDc0Yjc2MjI2ZTM0OTIwNjZiNDA1NGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l6DyPCUBmXm1ZXLiwtUt3rbLapBN5MEtjlfFx4DzYKw">
> > If this is going to take more than like an hour to figure out though, then I think we can skip them and open them as a separate issue and go ahead and merge this! ✅
> 
> I fixed part of this. Getting the thing to grow vertically, then scroll has been a bit problematic, but I'll spend a few more minutes. I agree it looks dumb.

Fixed!

magnew commented:
Render got messed up on this one and I had issues creating new ones. SO, I ran the e2e tests locally and I'm going to merge this.
